domain/application/service/metadata.go

```diff
@@ -472,6 +478,26 @@ func encodeMetadataScope(scope internalcharm.RelationScope) (charm.RelationScope
 	}
 }
 
+func verifyRelations(provides, requires, peers map[string]charm.Relation) error {
+	seenKeys := set.NewStrings()
+	for k := range provides {
+		seenKeys.Add(k)
+	}
+	for k := range requires {
+		if seenKeys.Contains(k) {
+			return errors.Errorf("%w %q", applicationerrors.CharmRelationNameConflict, k)
+		}
+		seenKeys.Add(k)
+	}
+	for k := range peers {
+		if seenKeys.Contains(k) {
+			return errors.Errorf("%w %q", applicationerrors.CharmRelationNameConflict, k)
+		}
+		seenKeys.Add(k)
```

Comment:
You can drop this addition, as we guarantee `peers` key to be unique.

Reply:
We could use a couple of intersections, but then we lose the errors being specific.

Reply:
Good point/ I've refactored